### DISCUSSION TITLE
Issue with Isar: "Class 'IsarImpl' has no instance getter 'ipersons'." (3.1.0+1)

### DISCUSSION THREAD

**Discussion by albrnick** [isar/isar#1364] [Help] [answered]
Hello! I know I must be doing something wrong. But I can't figure out what.  Right after opening Isar I try to access a collection of "Iperson", but it fails with the error: "Class 'IsarImpl' has no instance getter 'ipersons'." Here is the code in question

```
class IsarGameStorage  {
  static dynamic _isar;

  static Future initialize() async {
    final dir = await getApplicationDocumentsDirectory();
    _isar = await Isar.open(
      [IpersonSchema],
      directory: dir.path,
    );
    debugPrint('ipersons reference: ${_isar.ipersons}');
  }
}
```

The error is: `Unhandled Exception: NoSuchMethodError: Class 'IsarImpl' has no instance getter 'ipersons'.`

And even checking the iperson.g.dart file I see the line:

```
extension GetIpersonCollection on Isar {
  IsarCollection<Iperson> get ipersons => this.collection();
}
```

I'm using Isar: 3.1.0+1

I'm sure I must be doing something stupid, but I can't see it. :( Any help would be greatly appreciated. Thanks for your time.

**Comment by simc** [comment, marked as answer]
The problem is that you change the type of the Isar instance to dynamic. Change it to `static Isar _isar;` and it should work

**Reply by albrnick** [reply]
That worked perfectly! Thanks!! = )
